Content.Server/Tools/WelderSystem.cs

```diff
@@ -12,6 +16,39 @@ public class WelderSystem : EntitySystem
     {
         private readonly HashSet<WelderComponent> _activeWelders = new();
 
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<WelderComponent, SolutionChangedEvent>(OnSolutionChange);
+            SubscribeLocalEvent<WelderComponent, ExaminedEvent>(OnExamine);
+        }
+
+        private void OnExamine(EntityUid uid, WelderComponent component, ExaminedEvent args)
+        {
+            if (component.WelderLit)
+            {
+                args.Message.AddMarkup(Loc.GetString("welder-component-on-examine-welder-lit-message") + "\n");
+            }
+            else
+            {
+                args.Message.AddText(Loc.GetString("welder-component-on-examine-welder-not-lit-message") + "\n");
+            }
+
+            if (args.IsInDetailsRange)
+            {
+                args.Message.AddMarkup(Loc.GetString("welder-component-on-examine-detailed-message",
+                    ("colorName", component.Fuel < component.FuelCapacity / 4f ? "darkorange" : "orange"),
+                    ("fuelLeft", Math.Round(component.Fuel)),
+                    ("fuelCapacity", component.FuelCapacity)));
+            }
+        }
+
+        private void OnSolutionChange(EntityUid uid, WelderComponent component, SolutionChangedEvent args)
+        {
+            component.Dirty();
+        }
```

Comment:
I just had a thought, you can probably just put a public method on SolutionContainerSystem to UpdateAppearance on a particular solution.